File: plugins/TextPlugin/src/plugins/quorum/Libraries/Language/Types/Number.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package plugins.quorum.Libraries.Language.Types;

import quorum.Libraries.Language.Object$Interface;
import quorum.Libraries.Language.Types.Number$Interface;

/**
 * Number plugin for handling Number class functions for objects and primitives.
 * 
 * @author Melissa Stefik
 */
public class Number {
    public java.lang.Object $me = null;
    private double number = -1;
    
    public double GetMaximumValue(){
        return Double.MAX_VALUE;
    }
    
    public static double PrimitiveGetMaximumValue(){
        return Double.MAX_VALUE;
    }
    
    public double GetMinimumValue(){
        return Double.MIN_VALUE;
    }
    
    public static double PrimitiveGetMinimumValue(){
        return Double.MIN_VALUE;
    }
    
    public double GetNotANumberValue(){
        return Double.NaN;
    }
    
    public static double PrimitiveGetNotANumberValue(){
        return Double.NaN;
    }
    
    public double GetNegativeInfinityValue(){
        return Double.NEGATIVE_INFINITY;
    }
    
    public static double PrimitiveGetNegativeInfinityValue(){
        return Double.NEGATIVE_INFINITY;
    }
    
    public double GetPositiveInfinityValue(){
        return Double.POSITIVE_INFINITY;
    }
    
    public static double PrimitiveGetPositiveInfinityValue(){
        return Double.POSITIVE_INFINITY;
    }
    
    public double GetNumberOfBits(){
        return Double.SIZE;
    }
    
    public static int PrimitiveGetNumberOfBits(){
        return Double.SIZE;
    }
    
    public void SetValueNative(double value){
        number = value;
    }
    
    public static boolean PrimitiveEquals(double self, Object$Interface obj) throws quorum.Libraries.Language.Errors.Error{
        Number$Interface t = (Number$Interface)obj;
        return self == t.GetValue();
    }
    
    public static int PrimitiveGetInteger(double self){
        return (int)self;
    }
    
    public static String PrimitiveGetText(double self){
        return "" + self;
    }
    
    public static quorum.Libraries.Language.Support.CompareResult$Interface PrimitiveCompare(double self, quorum.Libraries.Language.Object$Interface obj){
        quorum.Libraries.Language.Support.CompareResult r = new quorum.Libraries.Language.Support.CompareResult();
        Number$Interface t = (Number$Interface)obj;
        if(self == t.GetValue()){
            r.result = r.EQUAL;
        } else if(self > t.GetValue()){
            r.result = r.LARGER;
        } else {
            r.result = r.SMALLER;
        }
        return r;
    }
    
    public String GetHex(){
        return java.lang.Double.toHexString(number);
    }
    
    public static String PrimitiveGetHex(double self){
        return java.lang.Double.toHexString(self);
    }
    
    public boolean IsInfinite(){
        return java.lang.Double.isInfinite(number);
    }
    
    public static boolean PrimitiveIsInfinite(double self){
        return java.lang.Double.isInfinite(self);
    }
    
    public boolean IsNotANumber(){
        return java.lang.Double.isNaN(number);
    }
    
    public static boolean PrimitiveIsNotANumber(double self){
        return java.lang.Double.isNaN(self);
    }
}

File: Quorum/Library/Standard/Plugins/python/Libraries/Language/Types/Number.py
def Number_(value):
	val = quorum_Libraries_Language_Types_Number_()
	val.SetValue__quorum_number(value)
	return val


class plugins_quorum_Libraries_Language_Types_Number_:
	def __init__(self, value = 0.0):
		self.number = value;
    
	def SetValueNative__quorum_number(self, value):
		self.number = value

	def GetMaximumValue(self):
		return float.fromhex('0x1.fffffffffffffp+1023')

	def GetMinimumValue(self):
		return float.fromhex('-0x1.fffffffffffffp+1023')

	def GetMinimumPositiveValue(self):
		return float.fromhex('0x1.0p-52')

	def GetPositiveInfinityValue(self):
		return float('inf')

	def GetNumberOfBits(self):
		return self.number.bit_length()

	def GetNegativeInfinityValue(self):
		return -float('inf')

	def GetNotANumberValue(self):
		return float('nan')

	def IsInfinite(self):
		return self.number == float('inf')

	def IsNotANumber(self):
		return self.number == float('nan')
		
	def GetHex(self):
		return hex(self.number)
		
	def GetHashCode(self):
		return self.number
